modules/core/src/main/java/org/apache/ignite/internal/managers/deployment/GridDeploymentClassLoader.java

```diff
@@ -596,9 +596,16 @@ private GridByteArrayList sendClassRequest(String name, String path) throws Clas
 
                 continue;
             }
+            
+            // Must not send to the participants that asked us for the resource.
+            // NOTE: We could probably assert this, since this should not be called in the
```

Comment:
The note here is a but ambiguous. Do you mean in the context of a resource request - i.e., a GridDeploymentRequest message?